### PR TITLE
add warnings labels for blocking assessment issues

### LINKED ISSUE #15738: A11y_AzureDataStudioSQLMigration_AzureSQLMigration_ScreenReader:NVDA is reading the link as learn more link which is not descriptive enough for the users when tabbing to the link.
[“Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

**GitHubTags:** #A11yMAS;#A11yTCS;#A11ySev2;#SH_AZUREDATASTUDIOSQLMigration_June2021;#SQL Azure Data Studio;#Win10;#FTP;#DesktopApp;#WCAG2.4.4;#AINotInScope;#NVDA;#JAWS;#Linkpurpose;#STP

### Environment Details:
Application Name: SQL Azure Data Studio
Windows Version: Win10
Screen Reader:NVDA,JAWS

### Repro Steps:

1. Launch the application-Azure Data Studio-Insiders and install extension Azure SQL Migration.
2. Connect to any sql server.
3. Tab till Azure SQL migration Tab and press enter.
4. Start Screen Reader:NVDA
5. Tab till learn more link using tab key.

### Actual:
NVDA is reading the link as learn more link which is not descriptive enough for the users when tabbing to the link.
**This issue is also observed with JAWS.**

### Expected:
NVDA should read as learn more things you need before migration link when tabbing to the link.

### User Impact:
Screen reader users will not understand the purpose of the link.

### PR DESCRIPTION
**Pull request by brian-harris**
This PR
* add descriptive label to Dashboard 'pre-requisites' page - fixes #15738
* fix assessment results title text
* adds warning banner to database assessment page when one or more databases has one or more migration blocking issues
* adds error icon and information tooltip next to blocking issues in issue list
* adjusts database selection page layout
* added workaround for DeclarativeTableComponent binding hang:  
   * issue #16445

Assessment with blocking issue:
image

Assessment without blocking issue:
image

Database selector page: 
image

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: "Code Scanning - Action"
```
Comment (brian-harris) [inline .github/workflows/codeql.yml:1]: unexpected change, will remove

File: extensions/sql-migration/src/constants/strings.ts [hunk 2]
```diff
@@ -230,6 +232,7 @@ export const RESOURCE_GROUP_CREATED = localize('sql.migration.rg.created', "Reso
 export const NAME_OF_NEW_RESOURCE_GROUP = localize('sql.migration.name.of.new.rg', "Name of new Resource group");
 // common strings
 export const LEARN_MORE = localize('sql.migration.learn.more', "Learn more");
+export const LEARN_MORE_ABOUT_PRE_REQS = localize('sql.migration.learn.more.pre.reqs', "Learn more about things you need before starting a migration.");
```
Comment (rchlkm) [inline extensions/sql-migration/src/constants/strings.ts:235]: NIT: did we run these texts with a content writer?
i.e. "Learn more about things" seems a bit casual?

**Review by goyal-anjali — APPROVED** [review]

File: extensions/sql-migration/src/constants/strings.ts (hunk 2 above)
Comment (brian-harris) [reply extensions/sql-migration/src/constants/strings.ts:235]: no good point, I improvised using the group heading.
image

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (aasimkhan30) [inline .github/workflows/codeql.yml:1]: Why is this file added to the PR?

**Review by aasimkhan30 — APPROVED** [review]
Please remove the codeql.yml from this PR

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (brian-harris) [reply .github/workflows/codeql.yml:1]: not sure how it showed up, or if it showed after a merge from main.  I would prefer to remove it.

File: extensions/sql-migration/src/constants/strings.ts (hunk 2 above)
Comment (brian-harris) [reply extensions/sql-migration/src/constants/strings.ts:235]: I confirmed the current text with the product owner is good.

**Review by rchlkm — APPROVED** [review]

File: extensions/sql-migration/src/wizard/databaseSelectorPage.ts [hunk 3]
```diff
@@ -280,14 +291,31 @@ export class DatabaseSelectorPage extends MigrationWizardPage {
 		return result;
 	}
 
+	// undo when bug #16445 is fixed
+	private createIconTextCell(icon: IconPath, text: string): string {
```
[the remark this answers is not in the document]
Comment (brian-harris) [reply extensions/sql-migration/src/wizard/databaseSelectorPage.ts:295]: that makes sense, the implementations are different in styling, but I can refactor it after #16445 is fixed and the commented code can be reverted.
